src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -3371,8 +3373,36 @@ class ArnoldGlobals
 			const IECore::Msg::Level level = \
 				( logmask == AI_LOG_DEBUG ) ? IECore::Msg::Level::Debug : g_ieMsgLevels[ min( severity, 3 ) ];
 
-			g_currentMessageHandler->handle( level, "Arnold", std::string( tabs, ' ' ) + msg_string );
+			std::stringstream msg;
+
+			const int flags = AiMsgGetConsoleFlags();
+			if( flags & AI_LOG_TIMESTAMP )
+			{
+				const boost::posix_time::time_duration elapsed = boost::posix_time::millisec( AiMsgUtilGetElapsedTime() );
+				msg << std::setfill( '0' );
+				msg << std::setw( 2 ) << elapsed.hours() << ":";
+				msg << std::setw( 2 ) << elapsed.minutes() << ":";
+				msg << std::setw( 2 ) << elapsed.seconds() << " ";
+			}
+			if( flags & AI_LOG_MEMORY )
+			{
+				const size_t mb = AiMsgUtilGetUsedMemory() / 1024 / 1024;
+				if( mb < 1024 )
+				{
+					msg << mb << "MB  ";
+				}
+				else
+				{
+					msg.setf( std::ios::fixed, std:: ios::floatfield );
+					msg << setprecision( 1 ) << ( float( mb ) / 1024.0f ) << "GB ";
```

Comment:
This is causing misalignment of the rest of the message when transitioning from Mb to Gb :

![image](https://user-images.githubusercontent.com/1133871/95352850-8d26c680-08ba-11eb-99be-73d9aef532c3.png)

Would be good to keep the width of the memory section constant regardless of the values involved.

Reply:
I was thinking about this, but as we don't have horizontal scrolling, I was wary of using up the space.

Reply:
In the example above, we're just talking about a single space aren't we? We know the number in MB is bounded to 4 characters, so we can at least pad the GB number to that too. That gets us up to 9999Gb before we need to give up and spill over into more space.

Reply:
A fine point indeed [see here](https://github.com/GafferHQ/gaffer/pull/3940/commits/73cd15cc570bc5732d86a8393425437e046ae984).